example/server/storage/client.go

```diff
@@ -185,7 +185,7 @@ func WebClient(id, secret string, redirectURIs ...string) *Client {
 		authMethod:                     oidc.AuthMethodBasic,
 		loginURL:                       defaultLoginURL,
 		responseTypes:                  []oidc.ResponseType{oidc.ResponseTypeCode},
-		grantTypes:                     []oidc.GrantType{oidc.GrantTypeCode, oidc.GrantTypeRefreshToken},
+		grantTypes:                     oidc.AllGrantTypes,
```

Comment:
i've thought about this change again... doesn't it mean we have a breaking change (in the client check)?

Reply:
Without this change the following grant types break:

- Device Authorization: was "fixed" on `main` in https://github.com/zitadel/oidc/pull/400, but this is not (yet) included on `next`. 
-  Token Exchange: will indeed have a breaking change. 

The other grant types where already checking the client:

- CodeExchange: https://github.com/zitadel/oidc/blob/f6cb47fbbb4685afec2c3033005a760e6b90b709/pkg/op/token_code.go#L55-L57
- RefreshToken: https://github.com/zitadel/oidc/blob/f6cb47fbbb4685afec2c3033005a760e6b90b709/pkg/op/token_refresh.go#L111-L113
- JWTProfile: doesn't take a client
- ClientCredentials: 
https://github.com/zitadel/oidc/blob/f6cb47fbbb4685afec2c3033005a760e6b90b709/pkg/op/token_client_credentials.go#L93-L95